mxcubeweb/routes/login.py

```diff
@@ -5,17 +5,13 @@
     request,
     jsonify,
     make_response,
-    redirect,
-    session,
 )
 from mxcubeweb.core.util import networkutils
 from flask_login import current_user
 
 
 def deny_access(msg):
-    resp = jsonify({"msg": msg})
-    resp.code = 409
-    return resp
+    return make_response(jsonify({"msg": msg}), 200)
```

Comment:
Yes,  the previous code was not working by the way, doing `resp.code=409` had no effect and the status code was always 200.

I was thinking of using 401, and I tried that but I then realized that we are so far (and maybe we should find a another way to do this), returning an error string to the client with a text message to be shown in the "error box" below the login fields (in the login form).

So, for the moment I kept the 200 and I made it explicit. I, as you, think that we should move away from this and use 401 or 403 instead. But that also means that we need to handle the error display in some other way.

We can perhaps handle that in the PR following this one, what do you think.

Reply:
Bad credentials (can not authenticate) is 401. If user is authenticated (good credentials) but does not have the permission to access resource, then it is 403.

Looking at the context where the `deny_access` function is used, I would argue that `deny_access` is not an appropriate name for this function. It is not the access per se that is denied, it is that the credentials are incorrect. At this point it is not a matter of denying or authorizing access. Maybe `reject_credentials` is a better name.

Reply:
Looking at the context where the `deny_access` function is used, I would argue that `deny_access` is not an appropriate name for this function. It is not the access per se that is denied, it is that the credentials are incorrect. At this point it is not a matter of denying or authorizing access. Maybe `reject_credentials` is a better name.

Reply:
But no worries with using 200 for now and improving error handling later of course! 👍

Reply:
Thought about this a bit more: while responding with 401 for the `/login` endpoint is the way to go (because the purpose of the request is to log in, so if it doesn't work, the server must inform the client with the appropriate error code), it doesn't necessarily make sense to do so for the `/login_info` endpoint.

The `/login_info` endpoint is not an authenticated endpoint; calling it when the user is logged out is valid and expected, so it shouldn't actually respond with 401 but just 200 and a body that conveys that no user is currently logged in (maybe an empty object).